manifests/init.pp

```diff
@@ -80,7 +80,9 @@
           description => $description
         }
       }
-      default: {}
+      default: {
+        fail("${value[frequency]} please choose one of hourly, daily, weekly, or monthly.")
```

Comment:
Failing will cause the entire class to quit execution, even if there may be other users that have valid frequencies for password rotation, that have not been rotated yet in this execution. I think logging with [`err`](https://www.puppet.com/docs/puppet/6/function.html#err) would be a better option

Reply:
And perhaps the message could be changed to `"'${value[frequency]}' is not valid. Please choose one of hourly, daily, weekly, or monthly".

Reply:
sure that sounds reasonable.